src/dmd/backend/elfobj.d

```diff
@@ -1500,8 +1500,9 @@ void Obj_compiler()
     enum n = compilerHeader.length;
     char[n + maxVersionLength] compiler = compilerHeader;
 
-    assert(config._version.length < maxVersionLength);
-    const newLength = n + config._version.length;
+    // _version is a null-terminated D string
+    assert(config._version.length + 1  < maxVersionLength);
+    const newLength = n + config._version.length + 1;
```

Comment:
I didn't even find a use for the null-terminator, as here it could be `compiler[n + newLength] = 0`.